pkg/devconsole/webhooks/webhooks.go

```diff
@@ -0,0 +1,184 @@
+package webhooks
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/openshift/console/pkg/auth"
+	"github.com/openshift/console/pkg/devconsole/common"
+)
+
+func CreateGithubWebhook(r *http.Request, user *auth.User) (common.DevConsoleCommonResponse, error) {
+	// POST /api/dev-console/webhooks/github
+	var request GithubWebhookRequest
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to parse request: %v", err)
+	}
+
+	bodyBytes, err := json.Marshal(request.Body)
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to marshal request body: %v", err)
+	}
+
+	GH_WEBHOOK_URL := fmt.Sprintf("%s/repos/%s/%s/hooks", request.HostName, request.Owner, request.RepoName)
+
+	var serviceRequest *http.Request
+	serviceRequest, err = http.NewRequest(http.MethodPost, GH_WEBHOOK_URL, bytes.NewReader(bodyBytes))
```

Comment:
Can you help me understand how auth is handled here? Is this endpoint not authenticated?

Reply:
Not sure if I understand the question correctly.
But if you're asking how users authenticate with GitHub to add a webhook to the repo, we send the Authorization header with a bearer token from the frontend (entered by the user). This token is included in the request body (bodyBytes). On the backend, we extract the headers, add them to our backend request, and then send it.